logger/options.go

```diff
@@ -40,43 +39,66 @@ type Options struct {
 	Timestamp     time.Time
 }
 
-func (e InvalidOptionMessage) String() string {
-	return fmt.Sprintf("Options.%s: %s", e.Option, e.Message)
+type fieldIssue struct {
+	field string
+	prob  string
 }
 
-func validateOptionLength(option string, value string, problems *[]string) {
-	if len(value) > 32 {
-		*problems = append(*problems, InvalidOptionMessage{option, "length must be less than 32"}.String())
+type optionsError struct {
+	issues []fieldIssue
+}
+
+func (e *optionsError) Error() string {
+	errorString := ""
+	for i := 0; i < len(e.issues); i++ {
+		errorString += fmt.Sprintf("%s: %s \n", e.issues[i].field, e.issues[i].prob)
+	}
+	return errorString
+}
+
+func validateOptionLength(option string, value string) fieldIssue {
+	if len(value) > 80 {
+		return fieldIssue{field: option, prob: "length must be less than 80"}
 	}
+	return fieldIssue{}
```

Comment:
I didn't realize you can only use `nil` with pointers. It seems a bit odd to me to be returning an empty struct (especially for successful operations). You might want to rethink how this works - perhaps returning a`*fieldIssue` instead. I'm not sure what the idiomatic approach is.

Reply:
updated!